packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -88,8 +97,66 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
     inputClassName,
     errorMessage,
     disabled,
-    children
+    children,
+    /* @conditional-compile-remove(mention) */
+    mentionLookupOptions
   } = props;
+  const inputBoxRef = useRef<HTMLDivElement>(null);
+
+  /* @conditional-compile-remove(mention) */
+  // Current suggestion list, provided by the callback
+  const [mentionSuggestions, setMentionSuggestions] = useState<Mention[]>([]);
+  /* @conditional-compile-remove(mention) */
+  // Current suggestion list, provided by the callback
+  const [activeSuggestionIndex, setActiveSuggestionIndex] = useState<number | undefined>(undefined);
+
+  /* @conditional-compile-remove(mention) */
+  // Index of the current trigger character in the text field
+  const [currentTriggerStartIndex, setCurrentTriggerStartIndex] = useState<number>(-1);
+  /* @conditional-compile-remove(mention) */
+  const [inputTextValue, setInputTextValue] = useState<string>('');
+  /* @conditional-compile-remove(mention) */
+  const [tagsValue, setTagsValue] = useState<TagData[]>([]);
+  /* @conditional-compile-remove(mention) */
+  const [selectionStartValue, setSelectionStartValue] = useState<number | null>(null);
+  /* @conditional-compile-remove(mention) */
+  const [selectionEndValue, setSelectionEndValue] = useState<number | null>(null);
+  /* @conditional-compile-remove(mention) */
+  const [shouldHandleOnMouseDownDuringSelect, setShouldHandleOnMouseDownDuringSelect] = useState<boolean>(true);
+
+  /* @conditional-compile-remove(mention) */
+  // Caret position in the text field
+  const [caretPosition, setCaretPosition] = useState<Caret.Position | undefined>(undefined);
+  /* @conditional-compile-remove(mention) */
+  // Index of where the caret is in the text field
+  const [caretIndex, setCaretIndex] = useState<number | null>(null);
```

Comment:
In ts/js we always prefer undefined if it is just initialize with no value (most case we just directly use undefined)